File: app/src/main/java/mishka/rssreader/data/model/RssItem.java
package mishka.rssreader.data.model;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.arch.persistence.room.TypeConverters;

import org.joda.time.DateTime;
import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;
import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.List;
import java.util.Locale;

@Entity
@TypeConverters(EnclosureListConverter.class)
@Root(name = "item", strict = false)
public class RssItem {
    @PrimaryKey(autoGenerate = true)
    private int id;

    @Element(name = "title", required = false)
    private String title;
    @Element(name = "link", required = false)
    private String link;
    @Element(name = "description", required = false)
    private String description;
    @Element(name = "pubDate", required = false)
    private String pubDate;
    @ElementList(name = "enclosure", required = false, inline = true)
    private List<Enclosure> enclosure;
    @Element(name = "full-text", required = false)
    private String fullText;

    @Root(name = "enclosure", strict = false)
    public static class Enclosure {
        @Attribute(name = "url")
        private String url;
    }

    public String getEnclosureLink() {
        if (enclosure.size() > 0)
            return enclosure.get(0).url;
        else return "";
    }


    public boolean hasImage() {
        if ((enclosure == null) || (enclosure.size() == 0))
            return false;

        return true;
    }


    public String getFormattedPubDate() {

        DateTimeFormatter dateTimeFormatter = DateTimeFormat.forPattern("EEE, dd MMM yyyy HH:mm:ss Z").withLocale(Locale.ENGLISH);

        DateTime dateTime = DateTime.parse(pubDate, dateTimeFormatter);

        String month = dateTime.monthOfYear().getAsText(new Locale("ru"));
        month = month.substring(0, 1).toUpperCase() + month.substring(1);
        String formattedPubDate = dateTime.toString("H:mm  |  d " + month);

        return formattedPubDate;
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getPubDate() {
        return pubDate;
    }

    public void setPubDate(String pubDate) {
        this.pubDate = pubDate;
    }

    public List<Enclosure> getEnclosure() {
        return enclosure;
    }

    public void setEnclosure(List<Enclosure> enclosure) {
        this.enclosure = enclosure;
    }

    public String getFullText() {
        return fullText;
    }

    public void setFullText(String fullText) {
        this.fullText = fullText;
    }
}

File: app/src/main/java/mishka/rssreader/data/model/RssItem.kt
package mishka.rssreader.data.model

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import android.arch.persistence.room.TypeConverters
import org.joda.time.DateTime
import org.joda.time.format.DateTimeFormat
import org.simpleframework.xml.Attribute
import org.simpleframework.xml.Element
import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root
import java.util.*
import kotlin.collections.ArrayList

@Entity
@TypeConverters(EnclosureListConverter::class)
@Root(name = "item", strict = true)
class RssItem {
    @PrimaryKey(autoGenerate = true)
    var id: Int? = null

    @get:Element(name = "title", required = false)
    @set:Element(name = "title", required = false)
    var title: String? = null

    @get:Element(name = "link", required = false)
    @set:Element(name = "link", required = false)
    var link: String? = null

    @get:Element(name = "description", required = false)
    @set:Element(name = "description", required = false)
    var description: String? = null

    @get:Element(name = "pubDate", required = false)
    @set:Element(name = "pubDate", required = false)
    var pubDate: String? = null

    @get:ElementList(name = "enclosure", required = false, inline = true)
    @set:ElementList(name = "enclosure", required = false, inline = true)
    var enclosure: List<Enclosure>? = null
        get() {
            return field ?: ArrayList()
        }

    @get:Element(name = "full-text", required = false)
    @set:Element(name = "full-text", required = false)
    var fullText: String? = null
        get() {
            return field ?: ""
        }


    fun getEnclosureLink(): String {
        val enclosure = this.enclosure
        if (enclosure != null) {
            if (enclosure.isNotEmpty())
                return enclosure[0].url!!
        }

        return ""
    }

    fun getFormattedPubDate(): String {
        val dateTimeFormatter = DateTimeFormat.forPattern("EEE, dd MMM yyyy HH:mm:ss Z").withLocale(Locale.ENGLISH)

        val dateTime = DateTime.parse(pubDate, dateTimeFormatter)

        var month = dateTime.monthOfYear().getAsText(Locale("ru"))
        month = month.substring(0, 1).toUpperCase() + month.substring(1)

        return dateTime.toString("H:mm  |  d " + month)

    }

    @Root(name = "enclosure", strict = false)
    class Enclosure {
        @field:Attribute(name = "url")
        var url: String? = null
    }

    override fun toString(): String {
        return "{$id]: $title $link $description $pubDate ${getEnclosureLink()} $fullText"
    }
}


